gui-agent/vmside.c

```diff
@@ -1976,13 +1979,32 @@ static pid_t do_execute_xorg(
     }
 }
 
-static void terminate_and_cleanup_xorg(Ghandles *g) {
+/* this cannot be a SIGCHLD signal handler because of signal safety */
+static void *xorg_waitpid_thread(void *p) {
+    Ghandles *g = p;
     int status;
+    waitpid(g->x_pid, &status, 0);
+    if (atomic_load(&terminating)) {
+        exit(0);
```

Comment:
Should the exit status really be zero here?

Reply:
This happens if we get SIGTERM. Not sure if exit(0) is what we should do in this case, but it is what the previous code did. What would you suggest?

Reply:
Not sure tbh.

Reply:
Maybe we would want to reset the SIGTERM sigaction and send SIGTERM to ourselves, so that the parent gets told we were killed by SIGTERM? Seems like unnecessary complexity though, if I'm being honest.